### PR TITLE
Update sysrepo, libyang, libnetconf2 and netopeer2 packages to latest versions

### PR DESCRIPTION
**Pull request by jsmolic**
Maintainer: me / @jsmolic (old maintainer is not working on this anymore, so I'll be taking over)
Compile tested: (ipq40xx, OpenWrt master)
Run tested: (ipq40xx, OpenWrt master)

Description: Update `sysrepo`, `libyang`, `libnetconf2` and `netopeer2` packages to latest versions

### CODE REVIEW

**Comment by neheb** [comment]
Works for me.

ping @lucize since he worked on libyang.

**Comment by rydy** [comment]
Hi,

I used the 'subscribe' command of the 'netopeer2-cli' tool to test the custom sysrepo plugin(*.so), I tested it in two different environments(Centos7.8 and OpenWRT), All third-party(netopeer2-1.1.39、libyang-1.0.184、libnetconf2-1.1.26 and sysrepo-1.4.70) libraries in both environments are the same version.

For Centos, no problems were found.
For OpenWRT, netopeer2-server occurs bus error.

netopeer2-cli log:
->subscribe --filter-xpath /o-ran-software-management:download-event
nc ERROR: Session 1: SSH channel poll error (Socket error: disconnected).
cli_send_recv: Failed to receive a reply.

netopeer2-server log:
[INF]: LY: Resolving unresolved data nodes and their constraints...
[INF]: LY: All data nodes and constraints resolved.
[INF]: LY: Resolving unresolved data nodes and their constraints...
[INF]: LY: All data nodes and constraints resolved.
[INF]: SR: Published event "rpc" "/notifications:create-subscription" with ID 1 priority 0 for 1 subscribers.
[INF]: SR: Processing "/notifications:create-subscription" "rpc" event with ID 1 priority 0 (remaining 1 subscribers).
Bus error

Can you give me some adivce?

Thanks,
Best regards.

**Comment by jsmolic** [comment]
> Hi Jakov,
> 
> I used the 'subscribe' command of the 'netopeer2-cli' tool to test the custom sysrepo plugin(*.so), I tested it in two different environments(Centos7.8 and OpenWRT), All third-party(netopeer2-1.1.39、libyang-1.0.184、libnetconf2-1.1.26 and sysrepo-1.4.70) libraries in both environments are the same version.
> 
Hi.
Hmm, I'm not really sure, I did not manage to reproduce this error, I tested the latest versions as well as the ones currently available. Did you try using updated packages from my branch?

**Comment by rydy** [comment]
> > Hi Jakov,
> > I used the 'subscribe' command of the 'netopeer2-cli' tool to test the custom sysrepo plugin(*.so), I tested it in two different environments(Centos7.8 and OpenWRT), All third-party(netopeer2-1.1.39、libyang-1.0.184、libnetconf2-1.1.26 and sysrepo-1.4.70) libraries in both environments are the same version.
> 
> Hi.
> Hmm, I'm not really sure, I did not manage to reproduce this error, I tested the latest versions as well as the ones currently available. Did you try using updated packages from my branch?

Yes, I have used the latest one. Can you follow the instructions in the email for the test?

File: libs/libnetconf2/patches/001-cmake_not_updated.patch [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-         message(FATAL_ERROR "LibSSH ${LibSSH_VERSION} includes regression bugs and libnetconf2 will NOT work properly, try to use another version")
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f21fec9..5b912af 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -148,7 +148,7 @@ endif()
+ # dependencies - libssh
+ if(ENABLE_SSH)
+     find_package(LibSSH 0.7.0 REQUIRED)
+-    if(LIBSSH_VERSION VERSION_EQUAL 0.9.3 OR LIBSSH_VERSION VERSION_EQUAL 0.9.4)
```
Comment (rydy) [inline libs/libnetconf2/patches/001-cmake_not_updated.patch:9]: there are problems with both 0.9.3 and 0.9.4, I am using libssh0.9.2 now, is that ok?

**Comment by jsmolic** [comment]
@rydy Sure, but I didn't receive an e-mail :)

File: libs/libnetconf2/patches/001-cmake_not_updated.patch (hunk 1 above)
Comment (jsmolic) [reply libs/libnetconf2/patches/001-cmake_not_updated.patch:9]: Yes.
I found the patch for libssh 0.9.4 that should solve the compatilibity problem, so I will add that to libssh, that should make 0.9.4 version usable with libnetconf2

**Review by rydy — COMMENTED** [review]
> @rydy Sure, but I didn't receive an e-mail :)

Have you received it by now?

**Comment by jsmolic** [comment]
@rydy I received it now, thanks

File: libs/libnetconf2/patches/001-cmake_not_updated.patch (hunk 1 above)
Comment (jsmolic) [reply libs/libnetconf2/patches/001-cmake_not_updated.patch:9]: Yes, I do now, I will test the patch I found for this

File: libs/libssh/patches/0004-channels-Avoid-returning-SSH_AGAIN-from-ssh_channel_.patch [hunk 2]
```diff
@@ -0,0 +1,41 @@
+From 6417f5a3cac8537ac6f6ff7fc1642dfaa0917fb4 Mon Sep 17 00:00:00 2001
+From: Jakub Jelen <jjelen@redhat.com>
+Date: Thu, 16 Apr 2020 11:13:34 +0200
+Subject: [PATCH] channels: Avoid returning SSH_AGAIN from
+ ssh_channel_poll_timeout()
+
```
Comment (jsmolic) [inline libs/libssh/patches/0004-channels-Avoid-returning-SSH_AGAIN-from-ssh_channel_.patch:6]: This addresses the issues mentioned in https://www.libssh.org/archive/libssh/2020-03/0000029.html and enables `libnetconf2` to work with `libssh-0.9.4`

@neheb Does this require a `PKG_RELEASE` bump for libssh?

Comment (neheb) [reply libs/libssh/patches/0004-channels-Avoid-returning-SSH_AGAIN-from-ssh_channel_.patch:6]: It's a code change so yes.

Comment (jsmolic) [reply libs/libssh/patches/0004-channels-Avoid-returning-SSH_AGAIN-from-ssh_channel_.patch:6]: Done

File: net/netopeer2/files/netopeer2-server-merge-hostkey.default [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+set -e
+
+# avoid problems with sudo path
+SYSREPOCFG=`which sysrepocfg`
+OPENSSL=`which openssl`
+
+# check that there is no SSH key with this name yet
+KEYSTORE_KEY=`$SYSREPOCFG -X -x "/ietf-keystore:keystore/asymmetric-keys/asymmetric-key[name='genkey']/name"`
+if [ -z "$KEYSTORE_KEY" ]; then
```
Comment (rydy) [inline net/netopeer2/files/netopeer2-server-merge-hostkey.default:11]: Is there a problem with outputting the following log information when the netopeer2-server start?
[INF]: LY: There is no leafref "/ietf-keystore:keystore/ietf-keystore:asymmetric-keys/ietf-keystore:asymmetric-key/ietf-keystore:name" with the value "genkey", but it is not required.

File: net/netopeer2/files/netopeer2-server-merge-hostkey.default [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+set -e
+
+# avoid problems with sudo path
+SYSREPOCFG=`which sysrepocfg`
+OPENSSL=`which openssl`
+
+# check that there is no SSH key with this name yet
+KEYSTORE_KEY=`$SYSREPOCFG -X -x "/ietf-keystore:keystore/asymmetric-keys/asymmetric-key[name='genkey']/name"`
```
Comment (rydy) [inline net/netopeer2/files/netopeer2-server-merge-hostkey.default:10]: Is there a problem with outputting the following log information when the netopeer2-server start?

[INF]: LY: There is no leafref "/ietf-keystore:keystore/ietf-keystore:asymmetric-keys/ietf-keystore:asymmetric-key/ietf-keystore:name" with the value "genkey", but it is not required.

**Pull request merged**